common/calculate-cpmm.ts

```diff
@@ -115,6 +119,53 @@ export function calculateCpmmPurchase(
   return { shares, newPool, newP, fees }
 }
 
+export function calculateCpmmAmount(
```

Comment:
I think there is probably a closed-form solution to find this amount...

Reply:
at the very least, there definitely are ways to compute it more efficiently if this ever becomes the bottleneck (unlikely).

Reply:
Ok, added a comment to that effect.